lib/index.js

```diff
@@ -64,34 +68,44 @@ const processQuerystring = (requestOptions, { applyToQueryString, valuesMap }) =
  */
 const pluginFactory = (params = {}) => {
   const pluginOptions = _.merge({}, defaultPluginOptions, params);
-
-  /**
-   * the actual plugin execution (runs per-request)
-   *
-   * @method load
-   * @param  {Object}   req            HTTP request object
-   * @param  {Object}   requestOptions options for executing HTTP request with oniyi-http-client
-   * @param  {Function} callback       [description]
-   * @example
-   * // simplified `req` and `requestOptions.uri` args
-   * load(null, { uri: '/api/foo/{ authType }/bar', authType: 'oauth' }, (err, requestOptions) => {
-   *  console.log(requestOptions.uri); // '/api/foo/oauth/bar'
-   * });
-   */
-  const load = (req, requestOptions, callback) => {
-    const { plugins: { formatUrlTemplate: pluginOptionsParam = {} } = {} } = requestOptions;
-    const options = _.merge({}, pluginOptions, pluginOptionsParam);
-
-    const uri = processUri(requestOptions, options);
-    const qs = processQuerystring(requestOptions, options);
-
-
-    callback(null, Object.assign({}, requestOptions, { uri, qs }));
-  };
+  debug('Plugin options compiled -> %O', pluginOptions);
 
   return {
-    name: 'format-url-template',
-    load,
+    name: PHASE_NAME,
+    onRequest: [{
+      phaseName: PHASE_NAME,
+      handler: (ctx, next) => {
+        const { options: requestOptions } = ctx;
+        const {
+          plugins: {
+            formatUrlTemplate,
+          } = {},
+          phasesToSkip: {
```

Comment:
seems like code that is better suited in `oniyi-http-client`... and there, actual plugin execution should be skipped.
If the plugin wants to provide `skip` switch via `options`, it should use a different name (e.g. `skipFormatUrlPlugin`). PhaseName must not be used. For now, I'd completely remove this feature. Feel free to provide a separate PR to add skipping logic

Reply:
hmm this is not what I had in mind. I agree that `skipping` logic could be part of `oniyi-http-client` though. But we must provide `phaseName` in order to skip specific phase hook handler for a specific phase list. 

A simple scenario is when a single plugin (like caching plugin) implements multiple phases(and also phase hook handler for each phase).  What if I want to skip only one of the phases, but not the whole plugin? Not to mention that plugin can implement phase hook handlers for both request / response phase lists, and I don't want to skip everything because of one phase, right?

Reply:
Another possible issue with switching logic of `skipping` into `oniyi-http-client` is when different phases within the same plugin need to communicate with each other through `hookState`... If I let http-client skip the phase hook handler, I have less control over `hookState`.

Reply:
so just use my suggestion. read prop from `options` and skip individual handler if needed.
Also, if you have multiple handlers, the each handler can set switches for subsequent handlers (e.g. the first one detects that the others are not needed)